packages/itwinui-css/src/table/sizes.scss

```diff
@@ -8,6 +8,11 @@
   &[data-iui-size='condensed'] {
     --_iui-table-header-size: calc(var(--iui-size-s) * 4);
     --_iui-table-row-size: calc(var(--iui-size-s) * 4 + 2px);
+
+    .iui-table-cell:not(:first-child) {
+      padding-left: calc(var(--iui-size-xs));
```

Comment:
Is only changing the padding of the cell enough?

I believe one of the main concerns in the original issue was that row expanders were taking up too much space.
![](https://user-images.githubusercontent.com/9084735/219112547-d9eb0e5e-cc8f-4e44-8c59-50f83dedc7f9.png)

Reply:
i think it can be lowered even more. this is what i see in the current version:
![](https://user-images.githubusercontent.com/9084735/220760516-268482c2-a370-4273-af00-ef9ecfd2ee71.png)

it's only a difference of 8px from the live version, probably because you only changed the "constant" part of the equation.

try changing the "dynamic" part as well
![](https://user-images.githubusercontent.com/9084735/220760986-a38f8979-80f1-49c7-afb2-fdf961183738.png)

Reply:
So I changed the multiplier for this based on the density. original 35 for default, 32 for condensed and 28 for extra-condensed. I saw the original author comment on how they got to 35 but I couldn't find where the expander_width + margin was set? Or how to manually calculate it? So I chose arbitrary multipliers that still look reasonably good/condensed.

Reply:
the items are misaligned now 
![](https://user-images.githubusercontent.com/9084735/221250612-8df270dc-cf22-4946-884b-31b2e41d03b2.png)

you might need to adjust the expander button's margin (yellow)
![](https://user-images.githubusercontent.com/9084735/221250922-ce366881-14e2-4bd4-bfea-10a4915daf05.png)

also it looks like the size of the first column is not going down (it's 48 regardless of density) :/
![](https://user-images.githubusercontent.com/9084735/221251138-7dd89f7e-abe6-4637-a4e1-189baf85f440.png)

Reply:
Regarding the first column/selection column not changing despite different density, it is working on my end by either: adding density to the return Table (in this specific case it looks like table Full 2 in Storybook), or by changing the density in Storybook control. How did you reproduce it?
![image](https://user-images.githubusercontent.com/22462084/221660323-49ee9751-c751-401a-b358-ec0a1f2e4141.png)

Reply:
see https://itwin.github.io/iTwinUI/1071/react/?path=/story/core-table--full-2&args=density:condensed

maybe you forgot to push some changes?

Reply:
It is working for me there too. But as mentioned, I have to click on another table and return back to Full 2 table to see the change, if you are using the control to toggle density.

Reply:
I have updated to the suggested padding, which fixes misalignments issue. Also added expandability to condensed table and accompanied Storybook visual test, as well as Table unit test per @gretanausedaite suggestion.